scripts/collect_ad_boilerplate.py

```diff
@@ -0,0 +1,200 @@
+
+        fp.close()
+
+    return pv_to_signal_mapping, include_file_base
+
+
+def write_cam_class(boilerplate_file, pv_to_signal_mapping, include_file_base, dev_name, det_name, cam_name):
+    """
+    Function that writes the boilerplate cam class. This includes the default configuration
+    attributes, along with all the attributes extracted from the template file.
+
+    This function is not perfect, as extracting a good attribute name from a PV name is not
+    always straightforward, since PV names are far from standard. Currently, the solution simply
+    adds an _ character before any capital letter, and then makes everything lowercase.
+    This works in most cases, but may require minor edits to certain names.
+
+    Examples:
+
+    EnableCallbacks -> enable_callbacks
+    EVTLoadGainFile -> e_v_t_load_gain_file
+    """
+
+    # Extend from FileBase class as well if necessary
+    file_base = ''
+    if include_file_base:
+        file_base = ', FileBase'
+
+    # Write class boilerplate
+    boilerplate_file.write(
+        f'''
+class {cam_name}(CamBase{file_base}):
+    _html_docs = ['{dev_name}Doc.html']
+    _default_configuration_attrs = (
+        CamBase._default_configuration_attrs
+    )
+''')
+
+    # Write appropriate attribute for
+    for pv in pv_to_signal_mapping.keys():
+        pv_name = pv
+        if pv_name.endswith('_RBV'):
+            pv_name = pv[:-4]
+        pv_var_name = re.sub('(?<!^)(?=[A-Z])', '_', pv_name).lower()
+        boilerplate_file.write(f"    {pv_var_name} = ADCpt({pv_to_signal_mapping[pv]}, '{pv}')\n")
+
+
+def parse_args():
+    """
+    usage: collect_ad_boilerplate.py [-h] [-t TARGET] [-d]
+
+    Utility for creating boilerplate ophyd classes
+
+    optional arguments:
+      -h, --help            show this help message and exit
+      -t TARGET, --target TARGET
+                            Location of locally installed areaDetector driver
+                            folder structure.
+      -d, --debug           Enable debug loogging for the script.
+    """
+
+    parser = argparse.ArgumentParser(description='Utility for creating boilerplate ophyd classes')
+    parser.add_argument(
+        '-t', '--target', help='Location of locally installed areaDetector driver folder structure.')
+    parser.add_argument('-d', '--debug', action='store_true', help='Enable debug loogging for the script.')
+
+    args = vars(parser.parse_args())
+
+    # Set logging level
+    log_level = logging.ERROR
+    if args['debug']:
+        log_level = logging.DEBUG
+
+    if args['target'] is None:
+        return os.path.abspath('.'), log_level
+    else:
+        return args['target'], log_level
+
+
+if __name__ == '__main__':
+
+    print()
+
+    # Check if input is valid
+    driver_dir, log_level = parse_args()
+    if not os.path.exists(driver_dir) or not os.path.isdir(driver_dir):
+        logging.error(f'Input {driver_dir} does not exist or is not a directory!')
+        exit(-1)
```

Comment:
What does a negative error code mean? Probably should use it from the `sys` library.